### DISCUSSION TITLE
Creating fields with islands?

### DISCUSSION THREAD

**Discussion by TheJoe1990** [Courseplay/Courseplay_FS22#1491] [Q&A]
Hey, I see there is a thing for "Islands" when you are generating the course.  I assume that in this context an "Island" would a be a big ass rock or tree in the middle of the field?

How do I generate such a field?

**Comment by 2terNick** [comment]
Hello,  courseplay then automatically recognizes when a field is interrupted by the ground and generates the course around the field island. For this, however, a field with a field island must be available.

**Reply by TheJoe1990** [reply]
> Hello, courseplay then automatically recognizes when a field is interrupted by the ground and generates the course around the field island. For this, however, a field with a field island must be available.

So, Im playing on No Mans Land, and I have been drawing custom borders for all the fields I want worked.  Are you saying that I have been doing unnecessary work and should have just let it look for a plowed area?  

Should I only be drawing custom fields when I need courseplay to do the plowing?  

Im sorry, Im new and trying my best to understand.

**Reply by 2terNick** [reply]
You need user-defined fields, on meadows that do not have a field no. have in play. Courseplay can automatically leave all fields with a number according to a specific pattern. A little tip for newcomers, there are nice tutorials on courseplay on youtube, take a look at a few to understand the basics of courseplay.

Benutzerdefinierte Felder benötigt man, auf Wiesen die keine Feld-Nr. im Spiel haben.  Alle Felder mit einer Nummer, kann courseplay automatisch nach einem bestimmten Muster abfahren.  Kleiner Tipp für Neueinsteiger, auf youtube gibt es schöne Tutorials zu courseplay, schau dir da mal ein paar von an, um die Basics von courseplay zu verstehen.

**Comment by Alkonik** [comment]
Тоже есть проблема с островком. обновил версию до последней. При создании маршрута активировал обход но что-то пошло не так.
https://steamcommunity.com/profiles/76561198152938192/screenshot/2399944328120140288/
Вернулся на внутриигровую версию. Теперь всё ок.
https://steamcommunity.com/profiles/76561198152938192/screenshot/2399944513743537302/

**Reply by Tensuko** [reply]
I am not sure if that field contour setting is working with island.
You can test without field contour?

**Reply by Alkonik** [reply]
Должна. ведь в настройках есть возможность . Внизу последние пункты. Версия 4.2 не увидела уже вспаханое и засеяное поле в 
 4.0 как поле с островом. Сейчас попробовал закрасить поле вокруг остова уже в 4.2 и да, остров виден.
![20240914185015_1](https://github.com/user-attachments/assets/3aa78384-ea8f-4692-a12c-f2ff9230eed4)

**Reply by Alkonik** [reply]
Всё понятно. Нужны танцы с бубном и другое сохранение где эти все курсы можно нарисовать и потом спокойно играть на основном)))
Как на счет идеи? Возможность рисовать поле не контуром а как в редакторе кистью? (на подобии как прописывают перемещение животных)

**Reply by Tensuko** [reply]
Drawing the field? Already possible, read ingame help.
But does not work with island.

**Reply by Alkonik** [reply]
Вот. А мысль в том чтоб поле рисовать можно было из CP сразу кистью. Тогда и острова были бы возможны. Но видимо это будет сложно... Не разобрался пока как на несколько орудий писать. Раньше был выбор количества орудий...

**Reply by Tensuko** [reply]
Sry I lost you totaly... can you try to translate it in english.... or @Gonimy-Vetrom may can help here.

However, why and what do you want to paint?
Islands can be paint, fields must use plow, always.
You can draw an outer field border no problem at all. Read ingame help how that works.

**Reply by Alkonik** [reply]
я в курсе как это работает. я о другом. Сейчас поле рисуется контуром... Вы в курсе как рисуют площадь для движения животных в загонах?

**Reply by Tensuko** [reply]
I don't know what you want to do....
If you talk about drawing the field border ingame in the building menu, have a look at the custom field editor.... Also in the ingame help menu.

**Reply by Tensuko** [reply]
![fsScreen_2024_09_15_18_20_15](https://github.com/user-attachments/assets/e2123a40-9180-4089-9b10-f2a6bd42aca6)

**Reply by Alkonik** [reply]
Я говорю про идею для курсплея. Чтоб рисовать поле не контуром а площадью. Как в редакторе через меню строительства. Например, на карте "no mans land" есть текстура пашни и ей можно нарисовать поле. Но не на всех картах такое есть и она рисует уже вспаханое поле. Я не сделать хочу. Я хочу чтобы Вы сделали или подумали как сделать.Это бы ,я думаю, упростило бы создание полей вообще. Я только спросил Вы в курсе как рисуют площадь для животных в редакторе (в модах)?Вот по такому же принципу можно было рисовать поля.

**Reply by Tensuko** [reply]
Now I got it....

Courseplay is not designed to do it that way.
This is something that have to be implemented in the map, as the texture have to be there.
For animals that is something totaly different then texturing fields and doing field density and stuff.

Sry but that won't work and is also not in our interest.

If you want to create your felds faster, I can suggest you this plow:
https://www.farming-simulator.com/mod.php?lang=de&country=de&mod_id=226965&title=fs2022

This one will do the job faster, but other than that, we won't do.
